Enable hypercore compression by default on the asset_datapoint table

Miggets7 opened the pull request:
Enables columnstore compression on the `asset_datapoint` table for storage savings and faster purges.

### Changes
- ***BREAKING Migration: Enable HyperCore with auto-compression policy (1 week) migration script will fail hard and fast if postgresql image doesn't support hypercore, ensure the postgres container is up to date***
- ***BREAKING Asset datapoint data is now retained infinitely by default and now uses weeks instead of days***
- Purging: Use drop_chunks() instead of DELETE to avoid decompression overhead on compressed chunks
- Utility: Add TimeUtil.toInstant() for datetime conversions
- Test: Validate compression and purge

### Benefits

- Significant storage reduction
- Instant chunk drops vs slow row-by-row deletes
- Improved query performance on historical data

Miggets7 committed:
Testing hypercore with purge test

Miggets7 committed:
Merge branch 'master' into feature/enabeling-hypercore

Miggets7 committed:
Make the test pass and catch the exception "tuple decompression limit exceeded by operation"

Miggets7 committed:
Delete whole chunks instead of decompressing them

Miggets7 committed:
Fix in test

Miggets7 committed:
Merge branch 'master' into feature/enabeling-hypercore

Miggets7 committed:
Added migration script for enabling HyperCore compression

Miggets7 committed:
Removed Thread sleep because the manual job run query waits for it to finish

Miggets7 committed:
Made hypercore interval configurable

denniskuijs committed:
Merge branch 'master' into feature/enabeling-hypercore

richturner committed:
Merge branch 'master' into feature/enabeling-hypercore

richturner reviewed (changes requested):
Going to want to change `DATA_POINTS_MAX_AGE_DAYS` to be something that aligns with DB chunks i.e. weeks as discussed in calls.

Will need a DB migration script to convert old DAYS to weeks

Miggets7 committed:
Merge branch 'master' into feature/enabeling-hypercore

Miggets7 committed:
Merge branch 'master' into feature/enabeling-hypercore

Miggets7 committed:
Merge branch 'master' into feature/enabeling-hypercore

wborn committed:
Merge branch 'master' into feature/enabeling-hypercore

Miggets7 committed:
Merge branch 'master' into feature/enabeling-hypercore

Miggets7 committed:
Merge branch 'master' into feature/enabeling-hypercore

# Conflicts:
#	manager/src/main/java/org/openremote/manager/datapoint/AssetDatapointService.java

Miggets7 committed:
Refactored to process feedback from PR

MartinaeyNL committed:
Merge branch 'master' into feature/enabeling-hypercore

Miggets7 committed:
Merge branch 'master' into feature/enabeling-hypercore

richturner committed:
Merge branch 'master' into feature/enabeling-hypercore

richturner commented on manager/src/main/java/org/openremote/manager/datapoint/AssetDatapointService.java:
```diff
@@ -197,6 +208,12 @@ protected void purgeDataPoints() {
                 });
             }
+        } catch (GenericJDBCException e) {
+            if (e.getSQLException().getSQLState().equals("53400") && e.getSQLException().getMessage().contains("tuple decompression limit exceeded by operation")) {
```
Seems brittle for the sake of a different log level just leave the previous exception catch but switch to `SEVERE` as this is a serious issue that needs addressing and shouldn't happen

Miggets7 committed:
Address PR review feedback for hypercore feature

- Simplify error handling in purgeDataPoints to single catch with SEVERE level
- Fix ClassCastException: handle both Instant and OffsetDateTime from TimescaleDB chunk metadata
- Hardcode columnstore policy interval to 7 days, remove configurable env var
- Reduce test dataset from 2.5M to 25K datapoints to keep test suite fast

Miggets7 committed:
Simplify datapoint purge to drop_chunks only with week-based retention

Replace complex per-attribute purge logic with a single drop_chunks call
aligned to 7-day chunk intervals. Switch from day-based to week-based
retention (default 105 weeks / ~2 years). Deprecate OR_DATA_POINTS_MAX_AGE_DAYS
and DATA_POINTS_MAX_AGE_DAYS meta item, keeping backward compatibility.

Miggets7 committed:
Merge branch 'master' into feature/enabeling-hypercore

# Conflicts:
#	manager/src/main/java/org/openremote/manager/datapoint/AssetDatapointService.java

Miggets7 committed:
Fix imports after merge with master

Add missing Asset import needed for matchesAssetIdPattern(),
remove unused Constants and Pattern imports from merge artifact.

Miggets7 committed:
Address PR review: purge respects custom retention, rename test

- Change purge disabled log message to be less misleading
- Purge now queries max custom DATA_POINTS_MAX_AGE_DAYS across all
  attributes and uses the greater of that or maxDatapointAgeWeeks
  as the effective drop_chunks cutoff
- Rename AssetDatapointHypercoreTest to AssetDatapointPurgeTest

Miggets7 committed:
Fix purge: schema-qualify drop_chunks table, use jsonb_exists

- Query hypertable schema from timescaledb catalog for drop_chunks
- Replace ? operator with jsonb_exists() to avoid Hibernate parameter conflict

Miggets7 committed:
Merge branch 'master' into feature/enabeling-hypercore

Miggets7 committed:
Merge branch 'master' into feature/enabeling-hypercore

richturner committed:
Merge branch 'master' into feature/enabeling-hypercore

richturner committed:
WIP

richturner committed:
Merge remote-tracking branch 'origin/master' into feature/enabeling-hypercore

richturner committed:
Merge branch 'master' into feature/enabeling-hypercore

richturner committed:
WIP

richturner committed:
WIP

richturner committed:
WIP

richturner committed:
Merge branch 'master' into feature/enabeling-hypercore

richturner committed:
WIP

richturner committed:
WIP

richturner committed:
WIP

richturner committed:
Merge branch 'master' into feature/enabeling-hypercore

richturner commented:
Need to decide what the fallback is if the postgres instance doesn't support hypercore, should configuration be part of DB image and then manager warns if `drop_chunks()` function is not available?

richturner committed:
Merge branch 'master' into feature/enabeling-hypercore

# Conflicts:
#	manager/src/main/java/org/openremote/manager/datapoint/AssetDatapointService.java

richturner committed:
WIP

richturner committed:
Robustness of hypercore migration script

richturner committed:
WIP

richturner committed:
Merge branch 'master' into feature/enabeling-hypercore

richturner committed:
WIP

richturner committed:
WIP

Miggets7 commented:
@richturner Looks good. I cannot approve it, because I'm the author

richturner commented:
The quick look over is fine, I will do the review of the code you wrote as well

richturner committed:
WIP

richturner committed:
WIP

richturner committed:
Merge branch 'master' into feature/enabeling-hypercore

richturner committed:
Merge branch 'master' into feature/enabeling-hypercore

richturner committed:
Merge branch 'master' into feature/enabeling-hypercore

richturner committed:
WIP

richturner committed:
WIP
